assembly/src/tokens/stream.rs

```diff
@@ -25,24 +26,40 @@ impl<'a> TokenStream<'a> {
             return Err(ParsingError::empty_source());
         }
         let mut tokens = Vec::new();
-        let mut doc_comments = BTreeMap::new();
+        let mut proc_comments = BTreeMap::new();
+        let mut module_comment = None;
 
-        let mut comment = Comment(None);
+        let mut comment_builder = CommentBuilder(None);
 
         for line in source.lines() {
             let line = line.trim();
             if line.starts_with(DOC_COMMENT_PREFIX) {
-                comment.append_line(line);
-            } else if line.is_empty() && !comment.is_empty() {
-                return Err(ParsingError::malformed_doc_comment(tokens.len()));
+                comment_builder.append_line(line);
+            } else if line.is_empty()
+                && !comment_builder.is_empty()
+                && tokens.is_empty()
```

Comment:
minor nit: I would swap the order of these two. If we check `tokens.is_empty()` first, it would short-circuit faster in most cases.

Also, shouldn't logic here be a bit more complex? For example:
```Rust
} else if line.is_empty() {
    if tokens.is_empty() && !comment_builder.is_empty() {
        if module_comment.is_none() {
            module_comment = comment_builder.take_comment();
        } else {
            // should be an error? a module comment is malformed?
        }
    }
    // this would skip the blank line
} else {
    ...
}
```